### PR TITLE
177618323 historical sol prices

### PR DESCRIPTION
**Pull request by kbiala**
#### What's this PR do?
- Create model, services and scripts to gather daily and historical sol prices.

#### How should this be manually tested?
- Run tests
- Run scripts from `script/sol_prices` (commands at the top of the files)
- Check after each scripts if `log/sol_prices.log` has been updated
- Check `sol_prices` table if records has been created

#### What are the relevant tickets?
- https://www.pivotaltracker.com/n/projects/2177859/stories/177618323

#### Task completed checklist:
- [] Is there appropriate test coverage?
- [] Did I take into consideration possible security vulnerabilities?
- [] Are the controllers secure?
- [] Was every important matter documented?

#### Things to watch out for
- [N] Does this PR have any migrations?
- [N] Does this PR add new dependencies?

### CODE REVIEW

File: test/logic/sol_prices/coin_gecko_logic_test.rb [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  test '#filter_volumes_by_date' do
+    vcr_cassette(@namespace, @vcr_name) do
+      p = Pipeline.new(200, @initial_payload)
+                  .then(&get_volumes_from_days)
+                  .then(&filter_volumes_by_date)
+
+      assert_not_nil p.payload[:volume]
+    end
+  end
+end
```
Comment (maciejkocylapc) [inline test/logic/sol_prices/coin_gecko_logic_test.rb:61]: all this tests only check if the values exists. Is this enough?

File: test/services/sol_prices/api_client_test.rb [hunk 2]
```diff
@@ -0,0 +1,4 @@
+module SolPrices
+  class ApiClientTest < ActiveSupport::TestCase
```
Comment (maciejkocylapc) [inline test/services/sol_prices/api_client_test.rb:2]: send_request method should be tested

**Review by maciejkocylapc — CHANGES_REQUESTED** [review]

File: app/services/sol_prices/api_wrappers/coin_gecko.rb [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+      def coins_list
+        @api_client.coins_list
+      end
+
+      def status
+        @api_client.status
+      end
+
+      def price(ids = @coin)
+        ids = ids
```
[the remark this answers is not in the document]
Comment (radek-polcode) [reply app/services/sol_prices/api_wrappers/coin_gecko.rb:34]: For current price.

File: app/services/sol_prices/api_wrappers/coin_gecko.rb [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+          days: days
+        )
+      end
+
+      def price_for_range(from:, to:)
+        # Convert dates to UNIX timestamp
+        from = from.to_time.to_i
+        to = to.to_time.to_i
+        options = default_options.slice(:vs_currency)
+        query = { from: from, to: to }.merge(options).to_query
```
[the remark this answers is not in the document]
Comment (radek-polcode) [reply app/services/sol_prices/api_wrappers/coin_gecko.rb:74]: Unused method, I removed.

File: script/sol_prices/coin_gecko_gather_historical_prices.rb [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+# 
+# RAILS_ENV=production bundle exec rails r script/sol_prices/coin_gecko_gather_historical_prices.rb 
+
+require_relative '../../config/environment'
+
+include SolPrices::CoinGeckoLogic
+include SolPrices::SharedLogic
+
+# First day when sol price appeared.
+datetime = DateTime.new(2020,7,27)
```
[the remark this answers is not in the document]
Comment (radek-polcode) [reply script/sol_prices/coin_gecko_gather_historical_prices.rb:14]: This is one time script to gather all available prices.

File: config/schedule.rb [hunk 6]
```diff
@@ -36,8 +36,10 @@
 set :output, File.join(Whenever.path, 'log', 'whenever.log')
 job_type :ruby_script,
          'cd :path && RAILS_ENV=:environment :bundle_bin exec :ruby_bin script/:task >> :whenever_path/log/:task.log 2>&1'
+job_type :ruby_script_sol_prices,
+         'cd :path && RAILS_ENV=:environment :bundle_bin exec :ruby_bin script/sol_prices/:task >> :whenever_path/log/:task.log 2>&1'
```
[the remark this answers is not in the document]
Comment (radek-polcode) [reply config/schedule.rb:40]: I am not sure what happens if the subfolder does not exists.

File: test/logic/sol_prices/coin_gecko_logic_test.rb (hunk 1 above)
Comment (radek-polcode) [reply test/logic/sol_prices/coin_gecko_logic_test.rb:61]: Changed

File: test/services/sol_prices/api_client_test.rb (hunk 2 above)
Comment (radek-polcode) [reply test/services/sol_prices/api_client_test.rb:2]: I think it's enough tested in api_wrappers tests where is used.

**Review by maciejkocylapc — APPROVED** [review]

**Pull request merged**